packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -0,0 +1,281 @@
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final List<Widget> children = <Widget>[];
+    children.add(
+      Container(
+        color: Colors.blue,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            Switch(
+              value: _unspecified,
+              onChanged: (final bool value) =>
+                  setState(() => _unspecified = !_unspecified),
+            ),
+            SizedBox(
+              width: 300, // TODO(monsieurtanuki): proper size
+              child: Text(
+                appLocalizations.nutrition_page_unspecified,
+                style: const TextStyle(color: Colors.white),
+                maxLines: 2,
+                overflow: TextOverflow.ellipsis,
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+    if (!_unspecified) {
+      children.add(
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            Text(appLocalizations.nutrition_page_per_100g),
+            Switch(
+              value: _perServing,
+              onChanged: (final bool value) =>
+                  setState(() => _perServing = !_perServing),
+            ),
+            Text(appLocalizations.nutrition_page_per_serving),
+          ],
+        ),
+      );
+      for (final OrderedNutrient orderedNutrient in _displayableList) {
+        final Widget? item = _getNutrientWidget(orderedNutrient);
+        if (item != null) {
+          children.add(item);
+        }
+      }
+      children.add(
+        ElevatedButton.icon(
+          onPressed: () async {
+            final List<OrderedNutrient> availables = <OrderedNutrient>[];
+            for (final OrderedNutrient orderedNutrient in _displayableList) {
+              final String id = orderedNutrient.id;
+              final dynamic value = _getValue(id);
+              final bool addAble = value == null && !orderedNutrient.important;
+              if (addAble) {
+                availables.add(orderedNutrient);
+              }
+            }
+            availables.sort(
+                (final OrderedNutrient a, final OrderedNutrient b) =>
+                    a.name!.compareTo(b.name!));
+            final OrderedNutrient? selected = await showDialog<OrderedNutrient>(
+                context: context,
+                builder: (BuildContext context) {
+                  final List<Widget> children = <Widget>[];
+                  for (final OrderedNutrient nutrient in availables) {
+                    children.add(
+                      ListTile(
+                        title: Text(nutrient.name!),
+                        onTap: () => Navigator.pop(context, nutrient),
+                      ),
+                    );
+                  }
+                  return AlertDialog(
```

Comment:
Did you check how it looks with a SmoothAlertDialog

Reply:
No I didn't, but I don't either if a `ListView` is a good choice: (at least in French) we start with all the "acids" and that's probably not a good idea. Maybe something like a text field that display all the matching nutrients instead.
But later.